elements/rh-navigation-item/rh-navigation-item.ts

```diff
@@ -0,0 +1,157 @@
+
+  @property({ reflect: true }) layout?: 'column' | 'row' | 'icon-only' = undefined;
+
+  @property({ reflect: true }) hide?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' = undefined;
+
+  @property({ reflect: true, type: Boolean }) standalone = false;
+
+  /** Shorthand for the `icon` slot, the value is icon name */
+  @property() icon?: IconNameFor<IconSetName>;
+
+  /** Icon set for the `icon` property - 'ui' by default */
+  @property({ attribute: 'icon-set' }) iconSet?: IconSetName;
+
+  @consume({ context, subscribe: true })
+  @property({ attribute: false })
+  private ctx?: RhNavigationItemContext;
+
+  /**
+   * Color palette
+   */
+  @colorContextProvider()
+  @property({ reflect: true, attribute: 'color-palette' }) colorPalette?: ColorPalette;
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    if (!isServer) {
+      this.#mo.observe(this, { attributeFilter: ['aria-current'], childList: true, subtree: true });
+    }
+  }
+
+  render() {
+    const { hide = '', layout = '', variant = '', standalone } = this;
+    const { compact = true } = this.ctx ?? {};
+    const classes = {
+      'highlight': !!this.#highlight,
+      'hide': !!this.hide,
+      [hide]: true,
+      [variant]: true,
+      [layout]: true,
+      'standalone': standalone,
+      compact,
+    };
+    return html`
+      ${this.variant === 'dropdown' ? html`
+        <details @toggle="${this.#detailsToggle}" class="${classMap(classes)}">
+          <summary>
+            ${this.layout !== undefined || this.standalone ? html`
+              ${this.icon ?
+                html`<rh-icon icon="${this.icon}" set="${ifDefined(this.iconSet)}"></rh-icon>`
+                : html`<slot name="icon"></slot>`}
+            ` : ``}
+            ${this.layout !== 'icon-only' ? html`<slot name="summary">${this.summary}</slot>` : html``}
+            ${this.layout !== undefined || this.standalone ? html`` : html`
+              <rh-icon icon="caret-down" set="microns"></rh-icon>
+            `}
+          </summary>
+          <rh-navigation-item-menu id="details-content" color-palette="lightest">
+            <slot></slot>
+          </rh-navigation-item-menu>
+        </details>
+      ` : html`
+        <div id="container" class="${classMap(classes)}">
+          <slot></slot>
+        </div>
+      `}
+    `;
+  }
+
+  #detailsToggle() {
+    this._open = this._details.open;
+    this.dispatchEvent(new RhNavigationItemEvent(this._open, this));
+  }
+
+  #mutationsCallback() {
+    // if the [aria-current-page] link is a child of the default slot ensure the state of the highlight
+    const [dropdownMenu] = this.#slots.getSlotted<HTMLElement>('');
+    this.#highlight = dropdownMenu.querySelector('[aria-current="page"]') ? true : false;
+    this.requestUpdate();
+  }
+
+  public close() {
+    this._details.open = false;
+  }
+
+  public open() {
```

Comment:
Should there be a declarative way to open a navigation item? 

I want to lean towards no as an answer, and this may be where this details summary implementation splits from an `rh-disclosure,` which can be set to `open` on page load by default.

I'm unsure of a use case where you would want the navigation to auto-load with a dropdown open declaratively. I think that might confuse a user when they do not trigger the interaction, as well it would "hide" content below it, unlike a disclosure, which pushes it down. 

I can see this behavior for jump links nav or an accordion/disclosure where you would potentially follow a #hash link to that section of the page, and you would want the disclosure open, but hiding page content with an open nav on load might be where my opinion splits.